template/web/app/paymaster-bundler/_components/GamePlay.tsx

```diff
@@ -1,31 +1,180 @@
+import { Dispatch, SetStateAction, useCallback, useState } from 'react';
+import { usePrivy } from '@privy-io/react-auth';
+import { ReloadIcon, SymbolIcon } from '@radix-ui/react-icons';
+import { useSpring, animated } from '@react-spring/web';
+import clsx from 'clsx';
+import { SmartAccountClient } from 'permissionless';
+import { PublicClient, encodeFunctionData } from 'viem';
+import { sepolia } from 'viem/chains';
+import Button from '@/components/Button/Button';
 import NextImage from '@/components/NextImage/NextImage';
+import createNFTMap from '../_utils/createNFTMap';
+import fetchNFTs from '../_utils/fetchNFTs';
+import { ALL_ITEMS } from '../constants';
+import { NFTType, OwnedTokensType } from '../types';
+import { nftAbi } from './abi';
+
+type GameplayProps = {
+  setOwnedTokens: Dispatch<SetStateAction<OwnedTokensType>>;
+  smartAccount?: SmartAccountClient;
+  client?: PublicClient;
+};
+
+const getRandomNumber = () => {
+  let randomNumber;
+  // Generate a random number between 0 and 3
+  const randomValue = Math.random();
+  if (randomValue < 0.25) {
+    // 25% chance of getting 0
+    randomNumber = 0;
+  } else {
+    // 75% chance of getting a number between 1 and 3
+    randomNumber = Math.floor(randomValue * 3) + 1;
+  }
+  return randomNumber;
+};
+
+export default function GamePlay({ setOwnedTokens, smartAccount, client }: GameplayProps) {
```

Comment:
The current GamePlay only works when logging in with an email account but not with Coinbase Wallet. Is that intentional? Is the wallet flow suppose to work too in this PR?

Reply:
removed, wasn't intended to allow users to use their wallet